### PR TITLE
Add static Server::start to avoid unnecessary inheritance

### PR DESCRIPTION
**Pull request by sergey-melnychuk**
Both `Server` and `DevelopmentServer` have no instance properties and introduce unnecessary inheritance. 
This could be substituted with static `start` method with optional explicit `additionalModules`.

### CODE REVIEW

File: testing/trino-server-dev/src/main/java/io/trino/server/DevelopmentServer.java [hunk 1]
```diff
@@ -14,30 +14,22 @@
...
-        extends Server
 {
     private DevelopmentServer() {}
 
-    @Override
-    protected Iterable<? extends Module> getAdditionalModules()
+    public static void main(String[] args)
     {
-        return ImmutableList.of(binder -> {
+        Server.start("dev", ImmutableList.of(binder -> {
```
Comment (ssheikin) [inline testing/trino-server-dev/src/main/java/io/trino/server/DevelopmentServer.java:29]: static import?

File: core/trino-server-main/src/main/java/io/trino/server/TrinoServer.java [hunk 2]
```diff
@@ -35,6 +35,6 @@ public static void main(String[] args)
         }
 
         String version = TrinoServer.class.getPackage().getImplementationVersion();
-        new Server().start(firstNonNull(version, "unknown"));
+        Server.start(firstNonNull(version, "unknown"));
```
Comment (ssheikin) [inline core/trino-server-main/src/main/java/io/trino/server/TrinoServer.java:38]: Static import?

File: core/trino-main/src/main/java/io/trino/server/Server.java [hunk 3]
```diff
@@ -72,12 +73,17 @@
 
 public class Server
 {
-    public final void start(String trinoVersion)
+    public static void start(String trinoVersion)
     {
-        new EmbedVersion(trinoVersion).embedVersion(() -> doStart(trinoVersion)).run();
+        new EmbedVersion(trinoVersion).embedVersion(() -> doStart(trinoVersion, Collections.emptyList())).run();
```
[the remark this answers is not in the document]
Comment (sergey-melnychuk) [reply core/trino-main/src/main/java/io/trino/server/Server.java]: done

File: testing/trino-server-dev/src/main/java/io/trino/server/DevelopmentServer.java (hunk 1 above)
Comment (sergey-melnychuk) [reply testing/trino-server-dev/src/main/java/io/trino/server/DevelopmentServer.java]: done

File: core/trino-server-main/src/main/java/io/trino/server/TrinoServer.java (hunk 2 above)
Comment (sergey-melnychuk) [reply core/trino-server-main/src/main/java/io/trino/server/TrinoServer.java]: done

File: core/trino-main/src/main/java/io/trino/server/Server.java [hunk 4]
```diff
@@ -72,12 +72,17 @@
 
 public class Server
 {
-    public final void start(String trinoVersion)
+    public static void start(String trinoVersion)
     {
-        new EmbedVersion(trinoVersion).embedVersion(() -> doStart(trinoVersion)).run();
+        new EmbedVersion(trinoVersion).embedVersion(() -> doStart(trinoVersion, ImmutableList.of())).run();
```
Comment (ssheikin) [inline core/trino-main/src/main/java/io/trino/server/Server.java:77]: reuse overloaded method here

Comment (sergey-melnychuk) [reply core/trino-main/src/main/java/io/trino/server/Server.java]: done

**Review by ssheikin — APPROVED** [review]

**Comment by sergey-melnychuk** [comment]
Looking into failure of `TestMySqlCaseInsensitiveMapping` (failed check).

**Comment by ssheikin** [comment]
TestMySqlCaseInsensitiveMapping is flaky https://github.com/trinodb/trino/issues/8477

**Comment by colebow** [comment]
:wave: @sergey-melnychuk - this PR has become inactive. If you're still interested in working on it, please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Pull request closed without merging**